dvc/lock.py

```diff
@@ -3,58 +3,57 @@
 from __future__ import unicode_literals
 
 import os
-import time
-import zc.lockfile
+from datetime import timedelta
+import hashlib
+from flufl.lock import Lock as _Lock, TimeOutError
 
 from dvc.exceptions import DvcException
+from dvc.utils import makedirs
 
 
 class LockError(DvcException):
     """Thrown when unable to acquire the lock for dvc repo."""
 
 
-class Lock(object):
+class Lock(_Lock):
     """Class for dvc repo lock.
 
     Args:
-        dvc_dir (str): path to the directory that the lock should be created
+        lockfile (str): the lock filename
             in.
-        name (str): name of the lock file.
+        lifetime (int | timedelta): hold the lock for so long.
+        tmp_dir (str): a directory to store claim files.
     """
 
-    LOCK_FILE = "lock"
     TIMEOUT = 5
 
-    def __init__(self, dvc_dir, name=LOCK_FILE):
-        self.lock_file = os.path.join(dvc_dir, name)
-        self._lock = None
+    def __init__(self, lockfile, lifetime=None, tmp_dir=None):
+        if isinstance(lifetime, int):
+            lifetime = timedelta(seconds=lifetime)
```

Comment:
Seems like these locks shouldn't have a finite `lifetime`, as we don't know for how long some command mind run. For some `dvc run`s it might be days or even weeks.

Reply:
Set lifetime to 30 days, you should adjust when you land your patch.